libdnf/dnf-repo.c

```diff
@@ -1215,12 +1354,20 @@ dnf_repo_check_internal(DnfRepo *repo,
         if (!dnf_repo_set_timestamp_modified(repo, error))
             return FALSE;
         age_of_data =(g_get_real_time() - priv->timestamp_modified) / G_USEC_PER_SEC;
-        if (age_of_data > permissible_cache_age) {
+
+        /*choose a lower value between cache and metadata_expire for expired checking */
```

Comment:
I could be wrong, but I believe  line:
`if (priv->kind != DNF_REPO_KIND_LOCAL && permissible_cache_age != G_MAXUINT)`
will be checked first and the code related to metadata_expire checking will be ignored by default.